chain/semver_test.go

```diff
@@ -0,0 +1,64 @@
+package chain
+
+import (
+	"testing"
+)
+
+func TestSemverCompatible(t *testing.T) {
+	testCases := []struct {
+		name     string
+		required semver
+		actual   semver
+		expected bool
+	}{
+		{
+			name:     "Equal Versions",
+			required: semver{Major: 1, Minor: 2, Patch: 3},
+			actual:   semver{Major: 1, Minor: 2, Patch: 3},
+			expected: true,
+		},
+		{
+			name:     "Different versions",
+			required: semver{Major: 1, Minor: 2, Patch: 3},
+			actual:   semver{Major: 2, Minor: 0, Patch: 0},
+			expected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := semverCompatible(tc.required, tc.actual)
+			if result != tc.expected {
+				t.Errorf("Expected: %v, got: %v", tc.expected, result)
```

Comment:
This isn't strictly required, but most of dcrd and other tools use a `got: %v, want: %v` idiom so you might want to get used to writing the checks like that.

Also, we _usually_ use `Fatalf`, unless you explicitly want the test to continue so you can perform additional assertions. Given that this is a subtest, doing a `Fatalf()` won't prevent the other tests from running.